src/overlays/actors/ovl_Bg_Iknv_Obj/z_bg_iknv_obj.c

```diff
@@ -21,32 +31,190 @@ const ActorInit Bg_Iknv_Obj_InitVars = {
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Iknv_Obj_0x80BD7AB0/func_80BD8040.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Iknv_Obj_0x80BD7AB0/BgIknvObj_UpdateType2.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Iknv_Obj_0x80BD7AB0/BgIknvObj_UpdateType1.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Iknv_Obj_0x80BD7AB0/BgIknvObj_UpdateDefault.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Iknv_Obj_0x80BD7AB0/BgIknvObj_Update.asm")
+static ColliderCylinderInit sCylinderInit = {
+    { 
+        COLTYPE_NONE, 
+        AT_NONE,
+        AC_ON | AC_TYPE_ENEMY, 
+        OC1_ON | OC1_TYPE_ALL, 
+        OC2_TYPE_1, 
+        COLSHAPE_CYLINDER, 
+    },
+    { 
+        ELEMTYPE_UNK0, 
+        { 0x00000000, 0x00, 0x00 }, 
+        { 0xF7CFFFFF, 0x00, 0x00 }, 
+        TOUCH_NONE | TOUCH_SFX_NORMAL, 
+        BUMP_ON, 
+        OCELEM_ON, 
+    },
+    { 40, 40, 0, { 0, 0, 0 } },
+};
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Iknv_Obj_0x80BD7AB0/BgIknvObj_Draw.asm")
+void BgIknvObj_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgIknvObj* this = THIS;
+    CollisionHeader* colHeader = NULL;
+
+    Actor_SetScale(&this->dyna.actor, 0.1f);
+    this->actionFunc = BgIknvObj_DoNothing;
+    switch (IKNV_OBJ_TYPE(this)) {
+        case IKNV_OBJ_WATERWHEEL: {
+            this->displayListPtr = D_06013058;
+            this->actionFunc = BgIknvObj_UpdateWaterwheel;
+            this->dyna.actor.flags |= 0x100000;
+            this->dyna.actor.flags |= 0x10;
+            return;
+        }
+        case IKNV_OBJ_RAISED_DOOR: {
+            this->displayListPtr = D_06011880;
+            BcCheck3_BgActorInit(&this->dyna, 0);
+            BgCheck_RelocateMeshHeader(&D_060119D4, &colHeader);
+            this->dyna.bgId = BgCheck_AddActorMesh(globalCtx, &globalCtx->colCtx.dyna, &this->dyna, colHeader);
+            this->actionFunc = BgIknvObj_UpdateRaisedDoor;
+            this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y + 120.0f;
+            return;
+        }
+        case IKNV_OBJ_SAKON_DOOR: {
+            this->displayListPtr = D_060129C8;
+            this->actionFunc = BgIknvObj_UpdateSakonDoor;
+            BcCheck3_BgActorInit(&this->dyna, 0);
+            BgCheck_RelocateMeshHeader(&D_06012CA4, &colHeader);
+            this->dyna.bgId = BgCheck_AddActorMesh(globalCtx, &globalCtx->colCtx.dyna, &this->dyna, colHeader);
+            Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit);
+            Collider_UpdateCylinder(&this->dyna.actor, &this->collider);
+            this->dyna.actor.colChkInfo.mass = 0xFF;
+            gSaveContext.weekEventReg[0x33] &= 0xEF;
+            Actor_SetHeight(&this->dyna.actor, IREG(88));
+            return;
+        }
+    }
+
+    Actor_MarkForDeath(&this->dyna.actor);
```

Comment:
```suggestion
            break;
        }
        default:
            Actor_MarkForDeath(&this->dyna.actor);
    }  
```
Would it still match if you change all the `return;` into `break;` and put `Actor_MarkForDeath` in `default:`?